### DISCUSSION TITLE
ifcopenshell c++  cmake

### DISCUSSION THREAD

**Discussion by 99zhu** [IfcOpenShell/IfcOpenShell#3308] [Q&A] [answered]
i meet some problem when cmake ifcopenshell c++.
who can give me a already cmaked ifcopenshell
thank you !

**Comment by aothms** [comment]
The idea of cmake is that you configure the project with the file paths on your system. The result is not portable. What errors do you run into?

**Comment by 99zhu** [comment]
the problem has there:
1.in Kernel.h #include don't work
![微信图片_20230614161735](https://github.com/IfcOpenShell/IfcOpenShell/assets/127181262/7519dceb-d5dc-458d-b665-979167ed8239)
2.in cmake 
![微信图片](https://github.com/IfcOpenShell/IfcOpenShell/assets/127181262/8c5bef3d-3333-4260-973e-db122a3782dd)
An error was reported during the compilation process of running "build-deps.cmd vs2019-x64 Debug". Select Ignore to have an impact on the compilation result？

**Reply by aothms** [reply, marked as answer]
1. See https://github.com/IfcOpenShell/IfcOpenShell/commit/840818d681ea3159948d7aff6e1a0ec17e69e785 You need to define the supported schemas at runtime. We use cmake for that. https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/cmake/CMakeLists.txt#L718 But you can also just use a `#define SCHEMA_SEQ (2x3)(4)`  in your own code.

For 2. I'm not sure. Do you also get this in RelWithDebInfo configuration? Ignore should be fine. We don't rely on any advanced functionality in HDF5.

**Reply by 99zhu** [reply]
thank you.the problem has already solved.
